cyral/internal/policyv2/model.go

```diff
@@ -0,0 +1,163 @@
+	normalizedDocument, err := normalizeJSON(r.Policy.Document)
+	if err != nil {
+		return fmt.Errorf("error normalizing 'document' field: %w", err)
+	}
+	if err := d.Set("document", normalizedDocument); err != nil {
+		return fmt.Errorf("error setting 'document' field: %w", err)
+	}
+
+	if err := d.Set("last_updated", map[string]interface{}{
+		"actor":      r.Policy.LastUpdated.Actor,
+		"actor_type": r.Policy.LastUpdated.ActorType,
+		"timestamp":  r.Policy.LastUpdated.Timestamp,
+	}); err != nil {
+		return fmt.Errorf("error setting 'last_updated' field: %w", err)
+	}
+	if err := d.Set("created", map[string]interface{}{
+		"actor":      r.Policy.Created.Actor,
+		"actor_type": r.Policy.Created.ActorType,
+		"timestamp":  r.Policy.Created.Timestamp,
+	}); err != nil {
+		return fmt.Errorf("error setting 'created' field: %w", err)
+	}
+	if err := d.Set("enforced", r.Policy.Enforced); err != nil {
+		return fmt.Errorf("error setting 'enforced' field: %w", err)
+	}
+	if r.Policy.Type != "" {
+		if err := d.Set("type", r.Policy.Type); err != nil {
+			return fmt.Errorf("error setting 'type' field: %w", err)
+		}
+	}
+	d.SetId(r.Policy.ID)
+	return nil
+}
+
+// ReadFromSchema reads the policy data from the schema
+func (r *PolicyV2) ReadFromSchema(d *schema.ResourceData) error {
+	r.Policy.ID = d.Get("id").(string)
+	r.Policy.Name = d.Get("name").(string)
+	r.Policy.Description = d.Get("description").(string)
+	r.Policy.Enabled = d.Get("enabled").(bool)
+	r.Policy.Tags = expandStringList(d.Get("tags").([]interface{}))
+	r.Policy.ValidFrom = d.Get("valid_from").(string)
+	r.Policy.ValidUntil = d.Get("valid_until").(string)
+
+	// Normalize the document field before storing it
+	document := d.Get("document").(string)
+	normalizedDocument, err := normalizeJSON(document)
+	if err != nil {
+		return fmt.Errorf("error normalizing 'document' field: %w", err)
+	}
+	r.Policy.Document = normalizedDocument
+
+	r.Policy.LastUpdated = expandChangeInfo(d.Get("last_updated").(map[string]interface{}))
+	r.Policy.Created = expandChangeInfo(d.Get("created").(map[string]interface{}))
+	r.Policy.Enforced = d.Get("enforced").(bool)
+	r.Policy.Type = d.Get("type").(string)
+	if r.Policy.Type == "" {
+		r.Policy.Type = "local"
+	}
+	return nil
+}
+
+// expandStringList converts a list of interface{} to a list of strings
+func expandStringList(list []interface{}) []string {
+	result := make([]string, len(list))
+	for i, v := range list {
+		result[i] = v.(string)
+	}
+	return result
+}
+
+// expandChangeInfo converts a map to a ChangeInfo struct
+func expandChangeInfo(m map[string]interface{}) ChangeInfo {
+	return ChangeInfo{
+		Actor:     getStringOrDefault(m["actor"]),
+		ActorType: getStringOrDefault(m["actor_type"]),
+		Timestamp: getStringOrDefault(m["timestamp"]),
+	}
+}
+
+// getStringOrDefault returns the string value or an empty string if nil
+func getStringOrDefault(v interface{}) string {
+	if v == nil {
+		return ""
+	}
+	return v.(string)
```

Comment:
This could panic if the actual value is not a string. Maybe you should return "" in the (unlikely) case that v is not a string?